src/logger.js

```diff
@@ -0,0 +1,62 @@
+const path = require('path');
+const winston = require('winston');
+const { app, remote } = require('electron');
+
+
+let userDataPath;
+let isDevMode;
+if (remote) {
+  // When this module is imported from render process, access via remote
+  userDataPath = remote.app.getPath('userData')
+  isDevMode = remote.process.argv[2] == '--dev'
+} else {
+  // But we also import it from the main process
+  userDataPath = app.getPath('userData')
+  isDevMode = process.argv[2] == '--dev'
+}
+
+/**
+ * Creates and returns a logger with Console & File transports.
+ *
+ * @param {string} label - for identifying the origin of the message
+ * @returns {logger} - with File and Console transports. 
+ */
+function getLogger(label, filename='log.txt') {
+  // Right now the logging in this app has multiple loggers streaming to the
+  // same file. Maybe that's okay? The goal is having only one file on disk.
+  // But it feels wrong to create several different loggers just so that
+  // each message can have a 'label' that refers to the calling file.
+  // The only alternative I know of is to pass extra metadata anytime we log,
+  // but that also feels wrong:
+  // `logger.debug('foo', {label: filename})`
```

Comment:
Having multiple loggers all stream to the same file is a _very_common use case.  We do it all over the place in InVEST, for example!  Of course there is some extra metadata that we could include in the logfile if we ever really needed to differentiate between the sources of the log messages.  In any case, I'd expect `winston` to be able to handle concurrent access in a way that makes sense.

Reply:
Great! In that case I pushed a change to use each file's filename as the logger's label to make reading the log a little easier.